easyopac_collection_links.module

```diff
@@ -130,23 +136,63 @@ function easyopac_collection_links_ding_entity_buttons($type, $entity) {
  * Generate preview link.
  *
  * @param array $preview
+ *   External resources array from LMS request.
+ * @param string $type
+ *   Requested item material type.
  *
  * @return mixed
  */
-function easyopac_collection_links_generate_preview_url(array $preview) {
+function easyopac_collection_links_generate_preview_url(array $preview, $type) {
+  $mtype = '';
   $preview_url = NULL;
+  $types = [
+    'ebook' => [
```

Comment:
This does not look right

Reply:
Maybe it looks wrong, but during debugging, development and testing I got responses from LMS where external resources was belonging to those types.

Reply:
What? What does these types define?

Reply:
In this array is defined mapping of material type (Bog, Ebog, Lydbog etc.) to the types of external resources (ebook, audiobook, videos).

Reply:
You mean on which types in LMS they are shown, or where they are available?

This is the match in LMS:
- ebook = face.type="ebog" AND facet.acSource="ereolen"
- audiobook = facet.type="lydbog (net)" AND facet.acSource="netlydbog"

Reply:
Which types in LMS they are shown. I'm working with `/details` response object (Ex: https://stg.lms.inlead.ws/hedbib/detail?faustNumber=47640962&withExternalResources) and the info I can use for correct mapping of a external resource link to EO material item can be taken only from `type` and `externalResources` keys.

Reply:
Sorry I'm not sure I understand. All item types in LMS can possibly contain these 2 values.